crates/binjs_io/src/multipart/write.rs

```diff
@@ -780,61 +780,53 @@ impl TokenWriter for TreeTokenWriter {
         );
         let index : TableIndex<_> = self.grammar_table.insert(description);
         {
+            debug!(target: "multipart", "tagged tuple index: {index:?}",
+                index = index);
+
             // Now, we determine if we should add an Offset
-            let inject_in_offset =
-                if let Some(&(_, ref tree)) = children.get(0)
-                {
-                    if let UnresolvedTree {
+            let mut children_data = vec![];
+            let mut is_lazy = false;
```

Comment:
The role of `is_lazy` isn't clear to me. You're trying to detect the field that comes just after a lazy field?

Reply:
lazy field is split into 2 children in the tree, the offset to skip the field, and the actual field.
then, the loop is trying to detect the offset field, and treat the next field as lazy field.

renamed `is_lazy_field`, and added comments.